flytekit/models/common.py

```diff
@@ -56,7 +56,7 @@ def __str__(self):
         return self.verbose_string()
 
     def __hash__(self):
-        return hash(self.to_flyte_idl().SerializeToString())
+        return hash(self.to_flyte_idl().SerializeToString(deterministic=True))
```

Comment:
@matthewphsmith is there a reason this wasn't here initially?

Reply:
ohh wow this looks like a bug? But, we never really use this right, we always use the git-sha as the version

Reply:
yeah git sha is used for task version, this is just used for equality.